lib/puppet/provider/logical_volume/lvm.rb

```diff
@@ -188,15 +188,17 @@ def size=(new_size)
 
             lvextend( '-L', new_size, path) || fail( "Cannot extend to size #{new_size} because lvextend failed." )
 
-            blkid_type = blkid(path)
-            if command(:resize4fs) and blkid_type =~ /\bTYPE=\"(ext4)\"/
-              resize4fs( path) || fail( "Cannot resize file system to size #{new_size} because resize2fs failed." )
-            elsif blkid_type =~ /\bTYPE=\"(ext[34])\"/
-              resize2fs( path) || fail( "Cannot resize file system to size #{new_size} because resize2fs failed." )
-            elsif blkid_type =~ /\bTYPE=\"(xfs)\"/
-              xfs_growfs( path) || fail( "Cannot resize filesystem to size #{new_size} because xfs_growfs failed." )
-            elsif blkid_type =~ /\bTYPE=\"(swap)\"/
-              swapoff( path) && mkswap( path) && swapon( path) || fail( "Cannot resize swap to size #{new_size} because mkswap failed." )
+            unless @resource[:resize_fs] == :false or @resource[:resize_fs] == false or @resource[:resize_fs] == 'false'
```

Comment:
One ruby shortcut for this is `unless [false, :false, 'false'].include?(@resource[:resize_fs])`